omc3/scripts/bad_bpms_summary.py

```diff
@@ -0,0 +1,294 @@
+"""
+Bad BPMs Summary
+----------------
+
+Scans all measurements in a list of given GUI output folders and compiles a list of bad BPMs with
+their given number of appearances after 'harpy' and 'isolation forest'.
+"""
+from collections.abc import Sequence
+from pathlib import Path
+
+import tfs
+from generic_parser import DotDict, EntryPointParameters, entrypoint
+
+from omc3.utils import logging_tools
+from omc3.utils.iotools import PathOrStr
+
+LOG = logging_tools.get_logger(__name__)
+
+# Constants ---
+ROOT = Path("/user/slops/data/LHC_DATA/OP_DATA/Betabeat")
+IFOREST = "IFOREST"
+HARPY = "HARPY"
+KEY = "HEADER_KEY"
+
+# Columns ---
+NAME = "NAME"
+ACCEL = "ACCELERATOR"
+PLANE = "PLANE"
+SOURCE = "SOURCE"
+REASON = "REASON"
+COUNT = "COUNT"
+FILE = "FILE"
+FILE_COUNT = "FILE_COUNT"
+PERCENTAGE = "PERCENTAGE"
+
+# Files ---
+MEASUREMENT_DIR = "Measurements"
+BAD_BPMS_HARPY = "*.bad_bpms_*"
+BAD_BPMS_IFOREST = "bad_bpms_iforest_*.tfs"
+
+
+def get_params():
+    params = EntryPointParameters()
+    params.add_parameter(
+        name="dates",
+        type=str,
+        nargs="+",
+        required=True,
+        help=("Dates to include in analysis. "
+        "This should be either subfolders in `root` or glob-patterns for those."
+        )
+    )
+    params.add_parameter(
+        name="root",
+        type=PathOrStr,
+        default=ROOT,
+        help="Path to the root directory, containing the dates."
+    )
+    params.add_parameter(
+        name="outfile",
+        type=PathOrStr,
+        help="Path to the file to write out." 
+    )
+    params.add_parameter(
+        name="print_percentage",
+        type=float,
+        help="Print out BPMs that appear in more than this percentage of measurements." 
+    )
+    params.add_parameter(
+        name="accel_glob",
+        type=str,
+        default="LHCB*",
+        help="Accelerator name (glob for the sub-directories)."
+    )
+    return params
+
+
+@entrypoint(get_params(), strict=True)
+def bad_bpms_summary(opt: DotDict):
```

Comment:
Is there any point in doing anything if opt.outfile and opt.print_percentage are both none? Could emit a warning or error?